lib/game/game_widget.dart

```diff
@@ -102,11 +113,15 @@ class _GameWidgetState extends State<GameWidget> {
   @override
   void initState() {
     super.initState();
+    // Add the initial overlays
+    widget.visibleOverlays?.forEach((key) {
+      widget.game.overlays.add(key);
```

Comment:
We need a way to add to this collection without firing listeners.

Reply:
yeah, that is why I am doing this before the listeners are registered